### DISCUSSION TITLE
How to apply Gumbel softmax to node features?

### DISCUSSION THREAD

**Discussion by gooaah** [pyg-team/pytorch_geometric#2742] [Q&A] [answered]
Hi everyone. I want to make continuous node features become one-hot before pooling but keep it differentiable (Gumbel softmax).

Is it implemented in the current version? If not, how to do it?

**Comment by rusty1s** [comment, marked as answer]
I think you can use the regular [`gumbel_softmax`](https://pytorch.org/docs/stable/generated/torch.nn.functional.gumbel_softmax.html#torch.nn.functional.gumbel_softmax) implementation of PyTorch for this.

**Reply by gooaah** [reply]
You are right, thanks!
